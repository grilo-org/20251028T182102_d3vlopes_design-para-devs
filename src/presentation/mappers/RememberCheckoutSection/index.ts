import { RememberCheckoutSectionAPI } from '@/api/models/Sections';
import { RememberCheckoutSectionModel } from '@/domain/models/Sections';

export class RememberCheckoutSectionMapper {
	static toDomain(
		api: RememberCheckoutSectionAPI,
	): RememberCheckoutSectionModel {
		return {
			id: api.sectionID,
			heading: api.title.html.replaceAll('<p></p>', ''),
			originalPrice: api.originalPrice,
			salePrice: api.salePrice,
			icons: api.icons.map((icon) => {
				return {
					...icon,
					src: icon.image.url,
					alt: String(icon.image.imageDescription),
				};
			}),
			button: {
				text: api.buttonText,
				href: api.buttonTarget,
			},
		};
	}
}
