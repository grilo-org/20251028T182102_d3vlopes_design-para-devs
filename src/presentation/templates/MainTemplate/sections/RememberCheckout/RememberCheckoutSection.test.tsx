import { render, screen } from '@testing-library/react';

import { rememberCheckoutSectionMock as mock } from './RememberCheckoutSection.mock';

import { RememberCheckoutSection } from '.';

const { getByRole, getByText, getByTestId } = screen;

jest.mock('@/presentation/components/atoms/Heading', () => ({
	Heading: () => <div data-testid="mock-heading" />,
}));

jest.mock('@/presentation/components/molecules/PaymentIcons', () => ({
	PaymentIcons: () => <div data-testid="mock-payment-icons" />,
}));

describe('<RememberCheckoutSection />', () => {
	beforeEach(() => render(<RememberCheckoutSection {...mock} />));

	it('should contain correct section id', () => {
		const { container } = render(<RememberCheckoutSection {...mock} />);

		const section = container.getElementsByTagName('section')[0];

		expect(section).toHaveAttribute('id', mock.id);
	});

	it('should render section heading', () => {
		const heading = getByTestId('mock-heading');

		expect(heading).toBeInTheDocument();
	});

	it('should render original price', () => {
		const originalPrice = getByText(/R\$ 60,00/i);

		expect(originalPrice).toBeInTheDocument();
	});

	it('should render sale price', () => {
		const salePrice = getByText(/R\$ 9,90/i);

		expect(salePrice).toBeInTheDocument();
	});

	it('should render button', () => {
		const button = getByRole('button', { name: mock.button.text });

		expect(button).toBeInTheDocument();
	});

	it('should render anchor with correct href value', () => {
		const anchor = getByRole('link');

		expect(anchor).toHaveAttribute('href', mock.button.href);
	});

	it('should render PaymentIcons', () => {
		const paymentIcons = getByTestId('mock-payment-icons');
		const text = getByText('Formas de pagamento');

		expect(paymentIcons).toBeInTheDocument();
		expect(text).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<RememberCheckoutSection {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
