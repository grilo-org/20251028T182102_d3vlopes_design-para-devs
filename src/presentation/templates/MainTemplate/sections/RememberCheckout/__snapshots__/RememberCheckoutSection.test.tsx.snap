// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RememberCheckoutSection /> should match snapshot 1`] = `
<div>
  <section
    class="flex flex-col items-center justify-center"
    id="remember-checkout"
  >
    <div
      data-testid="mock-heading"
    />
    <div
      class="flex flex-col items-center font-highlight font-bold my-8"
    >
      <span
        class="text-lg leading-[1.3613rem] text-complementary-red"
      >
        De
         
        <strong
          class="line-through font-bold"
        >
          R$ 
          60,00
        </strong>
      </span>
      <span
        class="text-2xl leading-[1.8156rem] text-complementary-white"
      >
        Por apenas
      </span>
      <strong
        class="text-[3rem] leading-[3.6306rem] text-complementary-green"
      >
        R$ 
        9,90
      </strong>
    </div>
    <a
      href="https://pay.hotmart.com/Y88426530P?checkoutMode=10&offDiscount=LANCAMENTO"
      rel="noreferrer"
      target="_blank"
    >
      <button
        class="font-heading inline-flex items-center justify-center whitespace-nowrap ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-gradient-primary rounded-[8px] py-6 font-bold tracking-wider w-[21.4375rem] text-xxl lg:w-[28rem] lg:text-5xl xl:w-[36.8125rem]"
      >
        Garantir desconto
      </button>
    </a>
    <div
      class="flex flex-col gap-3 items-center mt-6"
    >
      <p
        class="font-highlight font-bold text-neutral-100 mb-3 lg:text-lg"
      >
        Formas de pagamento
      </p>
      <div
        data-testid="mock-payment-icons"
      />
    </div>
  </section>
</div>
`;
